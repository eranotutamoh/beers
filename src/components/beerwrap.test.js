import React from 'react';
import { shallow } from 'enzyme';
import BeerWrap from './Beerwrap';
import BeerForm from './Beerform';
import BeerList from './beerlist/'

let beerWrapApp

beforeEach(() => beerWrapApp = shallow(<BeerWrap />))

it('renders without crashing', () => {
    shallow(<BeerWrap />)
});

it('should render a nav with text', () => {
    expect(beerWrapApp.find('nav a').text()).toEqual('Choose your beer');
});

it('should render a beer form', () => {
    const wrapper = shallow(
        <BeerForm params={{am: false, tax: false, weekend: true, food: ''}}
                  onClick={(param) => this.changeParameters(param)}
                  onChange={(foodvalue) => this.changeParameters('food', foodvalue)}
                  clearFood={() => this.changeParameters('food', '')}
        />
    );
    expect(wrapper).toMatchSnapshot();
});

it('should render a beer list', () => {
    const wrapper = shallow(
        <BeerList beers={[]}
                  loading={false}
        />
    );
    expect(wrapper).toMatchSnapshot();
});

/*it('handleNameInput', () => {
    let wrapper = shallow(<MyComponent/>);
    wrapper.instance().searchDish = jest.fn();
    wrapper.update();
    wrapper.instance().handleNameInput('BoB');
    expect(wrapper.searchDish).toBeCalledWith('BoB');
})*/

