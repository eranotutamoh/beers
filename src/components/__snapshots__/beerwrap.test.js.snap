// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a beer form 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="beer_form"
>
    <BeerQuestion
        onClick={[Function]}
        optionA="am"
        optionB="pm"
        parameter="am"
        status={false}
        title="A beer for AM or PM?"
    />
    <BeerQuestion
        onClick={[Function]}
        optionA="weekend"
        optionB="week-day"
        parameter="weekend"
        status={true}
        title="A beer for weekend or week-day?"
    />
    <BeerQuestion
        onClick={[Function]}
        optionA="yes"
        optionB="no"
        parameter="tax"
        status={false}
        title="Are you running tax reports?"
    />
    <section
        className="food"
    >
        <label>
            Enter a food
        </label>
        <input
            onChange={[Function]}
            placeholder="enter a food to match"
            type="text"
            value=""
        />
        <button
            onClick={[Function]}
        >
            Clear
        </button>
    </section>
</div>,
  "nodes": Array [
    <div
      className="beer_form"
>
      <BeerQuestion
            onClick={[Function]}
            optionA="am"
            optionB="pm"
            parameter="am"
            status={false}
            title="A beer for AM or PM?"
      />
      <BeerQuestion
            onClick={[Function]}
            optionA="weekend"
            optionB="week-day"
            parameter="weekend"
            status={true}
            title="A beer for weekend or week-day?"
      />
      <BeerQuestion
            onClick={[Function]}
            optionA="yes"
            optionB="no"
            parameter="tax"
            status={false}
            title="Are you running tax reports?"
      />
      <section
            className="food"
      >
            <label>
                  Enter a food
            </label>
            <input
                  onChange={[Function]}
                  placeholder="enter a food to match"
                  type="text"
                  value=""
            />
            <button
                  onClick={[Function]}
            >
                  Clear
            </button>
      </section>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BeerForm
        clearFood={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        params={
                Object {
                        "am": false,
                        "food": "",
                        "tax": false,
                        "weekend": true,
                      }
        }
/>,
      "_debugID": 9,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "clearFood": [Function],
          "onChange": [Function],
          "onClick": [Function],
          "params": Object {
            "am": false,
            "food": "",
            "tax": false,
            "weekend": true,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 5,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="beer_form"
>
          <BeerQuestion
                    onClick={[Function]}
                    optionA="am"
                    optionB="pm"
                    parameter="am"
                    status={false}
                    title="A beer for AM or PM?"
          />
          <BeerQuestion
                    onClick={[Function]}
                    optionA="weekend"
                    optionB="week-day"
                    parameter="weekend"
                    status={true}
                    title="A beer for weekend or week-day?"
          />
          <BeerQuestion
                    onClick={[Function]}
                    optionA="yes"
                    optionB="no"
                    parameter="tax"
                    status={false}
                    title="Are you running tax reports?"
          />
          <section
                    className="food"
          >
                    <label>
                              Enter a food
                    </label>
                    <input
                              onChange={[Function]}
                              placeholder="enter a food to match"
                              type="text"
                              value=""
                    />
                    <button
                              onClick={[Function]}
                    >
                              Clear
                    </button>
          </section>
</div>,
        "_debugID": 10,
        "_renderedOutput": <div
          className="beer_form"
>
          <BeerQuestion
                    onClick={[Function]}
                    optionA="am"
                    optionB="pm"
                    parameter="am"
                    status={false}
                    title="A beer for AM or PM?"
          />
          <BeerQuestion
                    onClick={[Function]}
                    optionA="weekend"
                    optionB="week-day"
                    parameter="weekend"
                    status={true}
                    title="A beer for weekend or week-day?"
          />
          <BeerQuestion
                    onClick={[Function]}
                    optionA="yes"
                    optionB="no"
                    parameter="tax"
                    status={false}
                    title="Are you running tax reports?"
          />
          <section
                    className="food"
          >
                    <label>
                              Enter a food
                    </label>
                    <input
                              onChange={[Function]}
                              placeholder="enter a food to match"
                              type="text"
                              value=""
                    />
                    <button
                              onClick={[Function]}
                    >
                              Clear
                    </button>
          </section>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BeerForm
    clearFood={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    params={
        Object {
            "am": false,
            "food": "",
            "tax": false,
            "weekend": true,
          }
    }
/>,
}
`;

exports[`should render a beer list 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="beer_list"
>
    <h2>
        0 beers for your perusal
    </h2>
</div>,
  "nodes": Array [
    <div
      className="beer_list"
>
      <h2>
            0 beers for your perusal
      </h2>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BeerList
        beers={Array []}
        loading={false}
/>,
      "_debugID": 13,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "beers": Array [],
          "loading": false,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 7,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="beer_list"
>
          <h2>
                    0 beers for your perusal
          </h2>
</div>,
        "_debugID": 14,
        "_renderedOutput": <div
          className="beer_list"
>
          <h2>
                    0 beers for your perusal
          </h2>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BeerList
    beers={Array []}
    loading={false}
/>,
}
`;
